src/main/scala/org/broadinstitute/dsde/agora/server/dataaccess/mongo/AgoraMongoClient.scala

```diff
@@ -15,4 +17,24 @@ object AgoraMongoClient {
   def getMethodsCollection(client: MongoClient): MongoCollection = {
     client(AgoraDatabase)(MethodsCollection)
   }
+
+  def getMongoClient: MongoClient = {
+    val server = new ServerAddress(Agora.server.mongoDbHost, Agora.server.mongoDbPort)
+
+    val user: Option[String] = Agora.server.mongoDbUser
+    val password: Option[String] = Agora.server.mongoDbPassword
+
+    (user, password) match {
+      case (Some(userName), Some(userPassword)) =>
+        val credentials = MongoCredential.createMongoCRCredential(
+          userName,
+          AgoraDatabase,
+          userPassword.toCharArray
```

Comment:
http://blog.earaya.com/blog/2012/07/25/using-char-arrays-for-passwords/

if you are curious